povm_toolbox/library/mutually_unbiased_bases_povm.py

```diff
@@ -0,0 +1,126 @@
+                transpile layout will be used as default value, 2) otherwise, a
+                simple one-to-one layout ``list(range(n_qubits))`` is used.
+            shot_repetitions: number of times the measurement is repeated for each
+                sampled PVM. More precisely, a new PVM is sampled for all ``shots``
+                (i.e. the number of times as specified by the user via the ``shots``
+                argument of the method :meth:`.POVMSampler.run`). Then, the parameter
+                ``shot_repetitions`` states how many times we repeat the measurement
+                for each sampled PVM (default is 1). Therefore, the effective total
+                number of measurement shots is ``shots`` multiplied by ``shot_repetitions``.
+            seed_rng: optional seed to fix the :class:`numpy.random.Generator` used to sample PVMs.
+                The Z-,X-,Y-measurements are sampled according to the probability distribution(s)
+                specified by ``bias``. The user can also directly provide a random generator. If
+                None, a random seed will be used.
+        """
+        if bias.shape[-1] != 3:
+            raise ValueError
+
+        if angles.shape == (3,):
+            processed_angles = self._process_angles(*angles)
+        elif angles.shape == (n_qubit, 3):
+            processed_angles = np.zeros((n_qubit, 6))
+            for i, angles_qubit_i in enumerate(angles):
+                processed_angles[i] = self._process_angles(*angles_qubit_i)
+        else:
+            raise ValueError
+
+        super().__init__(
+            n_qubit=n_qubit,
+            bias=bias,
+            angles=processed_angles,
+            measurement_twirl=measurement_twirl,
+            measurement_layout=measurement_layout,
+            shot_repetitions=shot_repetitions,
+            seed_rng=seed_rng,
+        )
+
+    @staticmethod
+    def _process_angles(theta: float, phi: float, lam: float) -> np.ndarray:
+        """Transform the three Euler angles into two angles for each of the rotated X,Y,Z measurements.
+
+        One way to obtain the rotated measurements would be to first (optionally) rotate the Z-measurement
+        into an X- or Y-measurement when applicable and then apply in all cases the fixed rotation defined
+        by ``theta``, ``phi`` and ``lam``. However, it means to have two subsequent rotations and therefore
+        two unitary gates are added to the circuits. Instead, we can look at the final orientation of the
+        rotated measurements and apply a direct rotation from the canonical Z-measurement to the respective
+        rotated measurements. Then, only one parametrized rotation gate is needed and two angles for each
+        rotated measurement.
+
+        Note: the sequence of intrinsic rotations is z-y'-z''.
+
+        Args:
+            theta: rotation around the y' axis.
+            phi: rotation around the z axis.
+            lam: rotation around the z'' axis.
```

Comment:
Did you accidentially change the order of arguments here?
Its correct in the docstring and further down you order them as `phi, theta, lam`. But when calling this method from `__init__`, they are input as `z-y'-z''` which would imply `phi, theta, lam`.

Please double check this.

Reply:
Yeah it is a bit of a mess and the docstrings are actually wrong.

I first followed the naming and order convention used in Qiskit with the `UGate(theta, phi, lam)`. So the `angles` in `__init__`  are expected in this order (I should document this explicitly). I chose this because a MUB is a LBCS-POVM to which we apply a `UGate(theta, phi, lam)` transformation and that's how we interpret it usually I think.

However the 3D rotation resulting from `UGate(theta, phi, lam)` in the Bloch sphere representation is a sequence `z-y'-z''` with angles `phi, theta, lam` resp.  That's why I changed the order internally...

Reply:
I have now updated the docstrings to better explain what I've just wrote above. Lmk if it's still unclear and confusing when reading it 😅